### DISCUSSION TITLE
model training issues

### DISCUSSION THREAD

**Discussion by Christcool** [ultralytics/ultralytics#24076] [Q&A]
Good morning !
 I am contacting you to report a serious problem that I encountered in the process of detecting plant disease with yolo.  The metrics are almost average, I tried to apply several techniques such as augmentation, contrast adjustment, resizing but despite everything the result remains critical.

I am contacting you because you certainly have the experience and if your time permits you will be able to help me! Thanks a lot

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Christcool, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you moving quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this appears to be a custom training ❓ question, please provide as much information as possible, including dataset image examples, your dataset YAML, model used, training command, key metrics, confusion matrix/results plots, and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 🌱
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Hi @Christcool, we can help, but low plant-disease metrics are usually caused by label quality, class imbalance, or a task mismatch more than augmentation alone; please attach 5–10 sample images with labels, per-class counts, your exact train command and starting weights, `results.png`/confusion matrix, and the output below, and if you recently fixed annotations delete any `labels/*.cache` files before retraining—our [YOLO troubleshooting guide](https://docs.ultralytics.com/guides/yolo-common-issues/) is also a useful checklist. If symptoms cover most of the leaf rather than small localized spots, `classify` or `segment` may fit better than `detect`.

```bash
yolo checks
```

**Reply by Christcool** [reply]
<img width="2250" height="1500" alt="BoxF1_curve" src="https://github.com/user-attachments/assets/94111ff9-548e-4731-a47d-5d6c96449dc3" />
<img width="2250" height="1500" alt="BoxP_curve" src="https://github.com/user-attachments/assets/8db5728f-3934-47cf-ba46-323f57116b6c" />
<img width="2250" height="1500" alt="BoxPR_curve" src="https://github.com/user-attachments/assets/fafa5a8c-7c76-45df-a499-81c701887773" />
<img width="2250" height="1500" alt="BoxR_curve" src="https://github.com/user-attachments/assets/50056f3f-df03-47cc-8c61-ea85e00d79c8" />
<img width="3000" height="2250" alt="confusion_matrix (1)" src="https://github.com/user-attachments/assets/d07efd5f-1bfa-4f79-99e0-800348e97259" />
<img width="2629" height="2365" alt="eda_visualisations" src="https://github.com/user-attachments/assets/700c5be7-6c92-40b4-bb9b-3a12d8b7889f" />
[results (5).csv](https://github.com/user-attachments/files/26384807/results.5.csv)

**Reply by Christcool** [reply]
these are a sample of my result

**Reply by Christcool** [reply]
@glenn-jocher

**Reply by glenn-jocher** [reply]
Thanks for the ping — the plots are useful, but to diagnose this I still need the last 10 rows of `results.csv` pasted as text, your exact `train` args or `args.yaml`, and 3–5 matching `val_batch*_labels.jpg`/`val_batch*_pred.jpg` pairs; if the confusion is mostly between similar disease classes, that usually points to class-definition or label-consistency issues more than augmentation, and the [performance metrics guide](https://docs.ultralytics.com/guides/yolo-performance-metrics/) may help interpret the curves.

**Reply by Christcool** [reply]
@glenn-jocher 
epoch,time,train/box_loss,train/cls_loss,train/dfl_loss,metrics/precision(B),metrics/recall(B),metrics/mAP50(B),metrics/mAP50-95(B),val/box_loss,val/cls_loss,val/dfl_loss,lr/pg0,lr/pg1,lr/pg2
1,1655.43,2.07852,2.85817,1.87781,0.32124,0.28629,0.21971,0.09519,2.0689,2.79163,1.86273,6.66496e-05,6.66496e-05,0.0667418
2,3262.33,1.98568,2.35214,1.76649,0.38211,0.34154,0.31997,0.13664,2.03934,2.27207,1.81993,0.000128917,0.000128917,0.0334708
3,4797.32,1.94298,2.18297,1.7279,0.41738,0.37279,0.35332,0.15914,1.97662,2.2111,1.75798,0.000186784,0.000186784,0.000195298
4,6348.03,1.91631,2.08472,1.70012,0.43147,0.39351,0.37882,0.17269,1.94539,2.1379,1.73359,0.0001802,0.0001802,0.0001802
5,7892.32,1.89635,2.02524,1.68051,0.44057,0.40406,0.38669,0.17661,1.94129,2.14161,1.72083,0.0001736,0.0001736,0.0001736
6,9442.48,1.88378,1.982,1.66583,0.45165,0.40078,0.39295,0.18182,1.92696,2.08837,1.70303,0.000167,0.000167,0.000167
7,10996.4,1.87101,1.9543,1.65304,0.45425,0.41339,0.4022,0.1866,1.91878,2.05356,1.69792,0.0001604,0.0001604,0.0001604
8,12548.8,1.8651,1.91983,1.64492,0.46587,0.41841,0.40873,0.18939,1.91566,2.02223,1.69028,0.0001538,0.0001538,0.0001538
9,14098.2,1.85629,1.8997,1.63642,0.46267,0.41922,0.40811,0.19019,1.91088,2.03281,1.68434,0.0001472,0.0001472,0.0001472
10,15649.2,1.84854,1.87871,1.62835,0.46817,0.42005,0.41148,0.19294,1.90672,2.02075,1.67666,0.0001406,0.0001406,0.0001406
11,17189.7,1.8437,1.86179,1.62263,0.47143,0.42226,0.41413,0.19418,1.90683,2.00369,1.67477,0.000134,0.000134,0.000134
12,18726.3,1.83707,1.84191,1.6161,0.47334,0.42149,0.41514,0.19523,1.90367,2.00604,1.67183,0.0001274,0.0001274,0.0001274
13,20269.7,1.82566,1.82624,1.61116,0.47848,0.42278,0.41676,0.19604,1.90215,1.99655,1.67031,0.0001208,0.0001208,0.0001208
14,21817.7,1.82439,1.81518,1.60888,0.47767,0.42586,0.41828,0.1964,1.90105,1.99369,1.66864,0.0001142,0.0001142,0.0001142
15,23368.8,1.82064,1.80426,1.60067,0.47929,0.42667,0.4186,0.19656,1.90045,1.99347,1.66676,0.0001076,0.0001076,0.0001076
16,24915.5,1.8163,1.79293,1.59585,0.48018,0.42588,0.41945,0.19721,1.89949,1.99012,1.6662,0.000101,0.000101,0.000101
17,26470,1.8123,1.78097,1.59331,0.47672,0.42879,0.42051,0.19767,1.89867,1.98519,1.66507,9.44e-05,9.44e-05,9.44e-05
18,28009.4,1.80579,1.76938,1.58936,0.47905,0.4289,0.42155,0.19825,1.89832,1.98172,1.66455,8.78e-05,8.78e-05,8.78e-05
19,29557,1.80335,1.75999,1.58495,0.47909,0.43136,0.4221,0.19847,1.89788,1.97699,1.66399,8.12e-05,8.12e-05,8.12e-05
20,31124.4,1.80078,1.74947,1.58169,0.47982,0.43136,0.42291,0.19877,1.89796,1.97239,1.66404,7.46e-05,7.46e-05,7.46e-05
21,32603.4,1.82458,1.67555,1.57912,0.48046,0.43031,0.42391,0.19932,1.89713,1.96833,1.66412,6.8e-05,6.8e-05,6.8e-05
22,34083.2,1.81651,1.65738,1.56938,0.48206,0.42892,0.42492,0.19987,1.89701,1.96345,1.66432,6.14e-05,6.14e-05,6.14e-05
23,35563.4,1.81284,1.64011,1.56501,0.48499,0.42621,0.42576,0.20035,1.89677,1.96106,1.66426,5.48e-05,5.48e-05,5.48e-05
24,37042.5,1.81023,1.62658,1.55947,0.48839,0.42535,0.42615,0.20078,1.89674,1.95729,1.66411,4.82e-05,4.82e-05,4.82e-05
![Uploading train_batch0.jpg.jpeg…]()

**Reply by glenn-jocher** [reply]
Thanks — from these logs `metrics/mAP50-95(B)` plateaus around `0.20` by epochs `20-24` while `train/*` and `val/*` losses flatten together, so this looks more like a data/task ceiling than an optimizer issue; the fastest next check is to overfit 20–50 carefully labeled images, and if it still cannot fit them nearly perfectly, relabel the hardest classes or switch to [classification](https://docs.ultralytics.com/tasks/classify/) or [segmentation](https://docs.ultralytics.com/tasks/segment/) if the disease affects most of the leaf rather than forming tight localized lesions.
